lib/command.ts

```diff
@@ -638,17 +638,31 @@ export class CommandInstance {
           // any new aliases need to be placed in positionalMap, which
           // is used for validation.
           if (!positionalMap[key]) positionalMap[key] = parsed.argv[key];
+
           // Addresses: https://github.com/yargs/yargs/issues/1637
-          // If both positionals/options provided,
-          // and no default or config values were set for that key,
-          // and if at least one is an array: don't overwrite, combine.
+          // In most cases, positionals will overwrite existing values on argv.
+          // They will be combined if both positionals/options are provided, and at least one is array type
           if (
-            !this.isInConfigs(yargs, key) &&
-            !this.isDefaulted(yargs, key) &&
             Object.prototype.hasOwnProperty.call(argv, key) &&
             Object.prototype.hasOwnProperty.call(parsed.argv, key) &&
             (Array.isArray(argv[key]) || Array.isArray(parsed.argv[key]))
           ) {
+            if (this.isInConfigs(yargs, key)) {
+              const shouldCombineArrays =
+                yargs.getInternalMethods().getParserConfiguration()[
+                  'combine-arrays'
+                ] === true;
+              if (shouldCombineArrays) {
+                // TODO: remove config values from argv[key] and parsed.argv[key], then re-add one set
```

Comment:
Instead of excluding keys with defaults/config from the combining logic, I would need to clean them up here and then combine them.

This part is tricky, as I can distinguish between user-provided values and default/config values in most cases (but not all).